src/Recommendations.js

```diff
@@ -33,30 +33,56 @@ const getRecommendedObject_TEMPORARY_BETA = (
     });
 };
 
+const scoredObjectID = (objectID, score) =>
+  `objectID:${objectID}<score=${score}>`;
+
 const buildSearchParamsFromRecommendations_TEMPORARY_BETA = (record, props) => {
-  let recoFilters = [];
-  let hitsPerPage = props.hitsPerPage;
+  let recoFilters;
+  let hitsPerPage;
+  const maxRecommendations = props.maxRecommendations || 0;
   const threshold = props.threshold || 0;
+  const fallbackFilters = props.fallbackFilters || [];
+  const hasFallback = fallbackFilters.length > 0;
 
   if (record.recommendations) {
     recoFilters = record.recommendations
       .reverse()
       .filter((reco) => reco.score > threshold)
-      .map(
-        (reco, i) =>
-          `objectID:${reco.objectID}<score=${Math.round(reco.score * 100) + i}>`
+      .map((reco, i) =>
+        scoredObjectID(reco.objectID, Math.round(reco.score * 100) + i)
       );
 
-    if (!hitsPerPage) {
-      hitsPerPage = record.recommendations.length;
+    // recommendations and fallback, force to retrieve maxRecommendations hits
+    if (hasFallback) {
+      hitsPerPage = maxRecommendations;
+    } else {
+      // otherwise max the hits retrieved with maxRecommendations
+      if (maxRecommendations > 0) {
+        hitsPerPage = Math.min(
+          record.recommendations.length,
+          maxRecommendations
+        );
+      } else {
+        hitsPerPage = record.recommendations.length;
+      }
+    }
+  } else {
+    recoFilters = [];
+
+    // no recommendations but fallback, force to retrieve maxRecommendations hits
+    if (hasFallback) {
+      hitsPerPage = maxRecommendations;
+    } else {
+      // otherwise, don't retrieve anything
+      hitsPerPage = 0;
     }
   }
```

Comment:
This is where the main change is. What do you think @Shipow @mthuret ?

Reply:
Works for me :)

Reply:
lgtm for now!
I'll explore a wrapper props options on another branch.